crates/starknet-server/src/api/http/models.rs

```diff
@@ -16,18 +18,46 @@ pub(crate) struct LoadPath {
 #[derive(Deserialize, Debug)]
 pub(crate) struct PostmanLoadL1MessagingContract {
     #[serde(rename = "networkUrl")]
-    network_url: String,
-    address: ContractAddress,
+    pub network_url: String,
+    pub address: Option<String>,
+
+    // TODO: THIS IS FOR TESTING PURPOSE ONLY.
+    #[serde(rename = "privateKey")]
+    pub private_key: String,
 }
 
 #[derive(Deserialize)]
 pub(crate) struct MessageToL2 {
-    l2_contract_address: ContractAddress,
-    entry_point_selector: EntryPointSelector,
-    l1_contract_addresss: ContractAddress,
-    payload: Calldata,
-    paid_fee_on_l1: Fee,
-    nonce: Nonce,
+    #[serde(rename = "l2ContractAddress")]
+    pub l2_contract_address: ContractAddress,
+    #[serde(rename = "entryPointSelector")]
+    pub entry_point_selector: EntryPointSelector,
+    #[serde(rename = "l1ContractAddress")]
+    pub l1_contract_address: ContractAddress,
+    pub payload: Calldata,
+    // TODO: using `u128` as type here will not allow hexadecimal strings. Is that ok?
+    // or do we prefer `Felt` to accept hexadecimal and then convert into `u128`?
+    #[serde(rename = "paidFeeOnL1")]
+    pub paid_fee_on_l1: u128,
```

Comment:
Do we want `u128` here, or we should prefer something that can accept "0x..." values?

Reply:
So this is for the `/postman/send_message_to_l1` endpoint (not governed by the spec)? If so, let's keep it as it was in Devnet-py so far, at least from the user's perspective - probably `Felt` makes the most sense (as your code comment says).

Why would it need to be `u128` internally? If it's really needed, we could use the custom serde deserialization annotation (next to the already used `rename`).

Reply:
`u128` it's the type that is used by `starknet_api` under the hood. IMHO, it's important to ensure that we are also bounded to this type to avoid too large values.

This is why in the first try I used `u128` because `axum` will automatically reject the request if it's too large.

But we can totally use `Felt` to accept more values like `hex`, and then we ensure we're not overflowing the `u128` value.

Reply:
Two things I care about are:
- keeping the API the same as it was in Devnet-py (meaning: accepting only hex strings for this fee param)
- not making it too complicated for future maintainers (e.g. having as few conversions)

If this can be done using `Felt` - great.

Reply:
@FabijanC I introduced a `U128HexOrDec` visitor for `serde`. I've tried a first implementation with `Felt`, but it not that obvious for the reader of the models to know that under the hood it's limited to `u128` values range.

With this `serde` visitor, we ensure that we can accept decimal or hexadecimal string, without compromising the readability of the model and the range check on the 

The related commit is this one: 90a70097659b478af5d86b22408391015141696a

please let me know if it makes sense to your too. If not, I'll be happy to revisit this.

Reply:
Since now decimal strings are also accepted, isn't this an expansion of the API? IMO having just one supported format is less error-prone.

Also, what is the advantage of introducing the new dependency (serde_with) when we can achieve this with existing dependencies? E.g. we already use `serialize_with`, `deserialize_with` and `with` in serde struct annotations.

Reply:
>Since now decimal strings are also accepted, isn't this an expansion of the API? IMO having just one supported format is less error-prone.
In that case, if you prefer to be 100% compatible with the devnet-py API, I'll use `Felt` as any other type. This is less correct for the model, but better for simplicity.

If it's not a problem to change a bit from devnet-py, we can use raw `u128` as `MintTokenRequest` is accepting.

>Also, what is the advantage of introducing the new dependency (serde_with) when we can achieve this with existing dependencies? E.g. we already use serialize_with, deserialize_with and with in serde struct annotations.

My bad on this, the habit of doing it with `serde_with` and the `Visitor`. Will remove it.

Reply:
> If this can be done using Felt - great.

To follow this, the commit 236ce995f707a06dbdc25316d35574726300a790 is working with `Felt` only. Let me know if it's ok that way.

Reply:
Looks ok, resolving